packages/contracts/src/governance/MainVotingPlugin.sol

```diff
@@ -199,6 +241,291 @@ contract MainVotingPlugin is Addresslist, MajorityVotingBase, IEditors, IMembers
+            to: _spacePlugin,
+            value: 0,
+            data: abi.encodeCall(SpacePlugin.acceptSubspace, (address(_subspaceDao)))
+        });
+        proposal_.actions.push(_action);
+
+        proposalCreators[proposalId] = msg.sender;
+
+        emit ProposalCreated({
+            proposalId: proposalId,
+            creator: proposalCreators[proposalId],
+            metadata: "",
+            startDate: _startDate,
+            endDate: proposal_.parameters.endDate,
+            actions: proposal_.actions,
+            allowFailureMap: 0
+        });
+
+        if (isEditor(proposalCreators[proposalId])) {
+            // Assuming that the proposer approves (if an editor)
+            vote(proposalId, VoteOption.Yes, false);
+        }
+    }
+
+    /// @notice Creates a proposal to make the DAO remove the given DAO as a subspace.
+    /// @param _subspaceDao The address of the DAO that holds the subspace to remove
+    /// @param _spacePlugin The address of the space plugin where changes will be executed
+    function proposeRemoveSubspace(IDAO _subspaceDao, address _spacePlugin) public onlyMembers {
+        if (address(_subspaceDao) == address(0) || _spacePlugin == address(0)) {
+            revert EmptyContent();
+        }
+        uint64 snapshotBlock;
+        unchecked {
+            snapshotBlock = block.number.toUint64() - 1; // The snapshot block must be mined already to protect the transaction against backrunning transactions causing census changes.
+        }
+        uint64 _startDate = block.timestamp.toUint64();
+
+        uint256 proposalId = _createProposalId();
+
+        // Store proposal related information
+        Proposal storage proposal_ = proposals[proposalId];
+
+        proposal_.parameters.startDate = _startDate;
+        proposal_.parameters.endDate = _startDate + duration();
+        proposal_.parameters.snapshotBlock = snapshotBlock;
+        proposal_.parameters.votingMode = votingMode();
+        proposal_.parameters.supportThreshold = supportThreshold();
+        IDAO.Action memory _action = IDAO.Action({
+            to: _spacePlugin,
+            value: 0,
+            data: abi.encodeCall(SpacePlugin.removeSubspace, (address(_subspaceDao)))
+        });
+        proposal_.actions.push(_action);
+
+        proposalCreators[proposalId] = msg.sender;
+
+        emit ProposalCreated({
+            proposalId: proposalId,
+            creator: proposalCreators[proposalId],
+            metadata: "",
+            startDate: _startDate,
+            endDate: proposal_.parameters.endDate,
+            actions: proposal_.actions,
+            allowFailureMap: 0
+        });
+    }
+
+    /// @notice Creates a proposal to remove an existing member.
+    /// @param _metadata The metadata of the proposal.
+    /// @param _member The address of the member who may eveutnally be removed.
+    function proposeRemoveMember(bytes calldata _metadata, address _member) public onlyMembers {
```

Comment:
Is there also a `proposeAddMember`? Or should we expect to call `proposeNewMember` on the member access plugin?

Reply:
Also I'm pretty sure that only editors should be able to propose to remove a member, but I posed this question in discord.

Reply:
@baiirun that's almost the same as what the MemberAccess plugin does. If you want to have a secondary way of doing the same, we can add it no problem

Changing the permission required for `proposeRemoveMember()` 👍

Reply:
Anybody should be able to able to request or propose to be a member, regardless of if they are a member or editor or not. I think that's what the Member Access Plugin already does right? 

I guess my question is if we should expose a wrapper for `proposeNewMember` in the main voting plugin and if that even makes sense.

Reply:
Same question here.
- If we want anyone to request membership without polluting the list of proposals with spam, then we need a separate plugin.
- If you still want another way of doing the same on the main plugin but with a filter (only members), then we can add it to the MainVoting plugin. 
  - However, you need to know an existing member to propose you. And I wouldn't personally recommend having 2 different ways of doing the same thing.

Reply:
We filter it in the substream already. We post some metadata with each proposal on IPFS that signals the "type" of the proposal, whether it's edits, a member request, etc. These types get stored in the substream and we query in the UI based on the type of proposal we want to view.

The main reason I ask is that it needs to be obvious for third-party developers how they should do contract interactions, and right now there's several contracts they have to interact with. It would be simpler if there was just one contract to interact with. So if you're in a space with governance you know that you really only ever interact with the main voting plugin. If you're in a personal space you only ever interact with the personal admin plugin. Not sure if that's easily achievable.